kubernetes/client_factory.go

```diff
@@ -219,10 +219,10 @@ func (cf *clientFactory) newClient(authInfo *api.AuthInfo, expirationTime time.D
 		}
 
 		// Replace the Kiali SA token with the user's auth token.
-		// In anonymous mode and when OpenID RBAC is disabled use the Kiali SA token
-		// since these modes don't have a user token.
-		if (cfg.Auth.Strategy != kialiConfig.AuthStrategyAnonymous) ||
-			(cfg.Auth.Strategy == kialiConfig.AuthStrategyOpenId && cfg.Auth.OpenId.DisableRBAC) {
+		// Only if we are not in an anonymous mode
+		// and in case we use OpenID with RBAC is enables.
+		if (cfg.Auth.Strategy != kialiConfig.AuthStrategyAnonymous) &&
+			(cfg.Auth.Strategy == kialiConfig.AuthStrategyOpenId && !cfg.Auth.OpenId.DisableRBAC) {
```

Comment:
The conditional (that I wrote) is kind of confusing here. We only want to use the Kiali SA token when 1. anonymous mode is enabled or 2. openid + rbac disabled. However the config returned from `cf.getConfig` has the SA client token already set so we need to replace with the user token in all but those two cases. That said, this now looks correct since we don't need to evaluate the right side if we're in anonymous mode and then only use the user token when openid is set and rbac is not disabled.

Reply:
Added tests